fms/datasets/__init__.py

```diff
@@ -138,3 +138,25 @@ def __iter__(self):
                 next_val = self.buffer[: self.max_seq_len]
                 self.buffer = self.buffer[self.max_seq_len :]
                 yield next_val
+
+
+class WithSeparatorDataset(Dataset, SavableDataset):
+    def __init__(
+        self,
+        dataset: SavableDataset,
+        bos_token_id: Optional[int] = None,
+        eos_token_id: Optional[int] = None,
+    ):
+        self.dataset = dataset
+        self.bos_token_id = bos_token_id
+        self.eos_token_id = eos_token_id
+
+    def __iter__(self):
```

Comment:
interesting model.. leave the original unmolested.. add tokens as needed in the view..

wonder if this pattern works, performance wise, in python particularly if for example one is going to operate on the entire range.. 

..c/c++/java/go primarily systems engineer getting used to python patterns